server/routes.js

```diff
@@ -0,0 +1,166 @@
+      res.send(err.message).status(500)
+    })
+}
+
+function getRecipient(req, res) {
+  var recipientId = req.params.id
+  db.getRecipient(recipientId)
+    .then(function (recipient) {
+      res.json(recipient)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+function updateTicket(req, res) {
+  var ticket = {
+    id: req.body.ticketId,
+    actualKg: req.body.actualKg,
+  }
+  db.updateTicket(ticket)
+    .then(function () {
+      res.json(ticket)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+function updateComment(req, res) {
+  var comment = {
+    ticketId: req.body.ticketId,
+    comment: req.body.comment
+  }
+  db.updateComment(comment)
+    .then(function () {
+      res.json(comment)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+// "has to match table column name, use _": req.body.useCamelCase
+  function addTicket(req, res) {
+    console.log(req.body.recipientId)
+    if (!req.body.recipientId) {
+      var ticket = {
+        expected_kg: req.body.expectedKg,
+        donor_id: req.body.donorId.split('|')[0],
+        details_id: req.body.donorId.split('|')[1],
+        is_complete: 0
+      }
+    } else {
+      var ticket = {
+        expected_kg: req.body.expectedKg,
+        recipient_id: req.body.recipientId.split('|')[0],
+        details_id: req.body.recipientId.split('|')[1],
+        is_complete: 0
+      }
+    }
+    console.log(ticket)
+    db.addTicket(ticket)
+      .then(function () {
+        res.json(ticket)
+      })
+      .catch(function (err) {
+        res.send(err.message).status(500)
+      })
+  }
+
+function getTickets(req, res) {
+  db.getTickets()
+    .then(function (tickets) {
+      res.json(tickets)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+function getDonorTicket(req, res) {
+  var ticketId = req.params.id
+  db.getDonorTicket(ticketId)
+    .then(function (donorTicket) {
+      res.json(donorTicket)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+function getRecipientTicket(req, res) {
+  var ticketId = req.params.id
+  db.getRecipientTicket(ticketId)
+    .then(function (recipientTicket) {
+      res.json(recipientTicket)
+    })
+    .catch(function (err) {
+      res.send(err.message).status(500)
+    })
+}
+
+function getTicketComments(req, res) {
+  var ticketId = req.params.id
+  db.getTicketComments(ticketId)
+    .then(function (ticketComments) {
+      var singleTicketComments = []
+      var foundComments = []
+      for (var i = 0; i < ticketComments.length; i++) {
+        if (!foundComments.includes(ticketComments[i].comments)) {
+          singleTicketComments.push({
+            comments: ticketComments[i].comments
+          })
+          foundComments.push(ticketComments[i].comments)
+        }
+      }
+      res.json(singleTicketComments)
```

Comment:
This route was a bit of a WTF moment for me :) I actually have no idea what it's doing. `db.getTicketComments` already retrieves the comments for a single ticket, so I'm not sure why all the additional processing?